Ical.Net/Utility/DateUtil.cs

```diff
@@ -19,7 +19,7 @@ public static IDateTime StartOfDay(IDateTime dt)
         => dt.AddHours(-dt.Hour).AddMinutes(-dt.Minute).AddSeconds(-dt.Second);
 
     public static IDateTime EndOfDay(IDateTime dt)
```

Comment:
We should probably get rid of that. Calculations should be done precisely, adding -1 whatever shouldn't be necessary.

Reply:
Created issue #662 for a separate PR